### PR TITLE
Support None as op input in GraphBuilder

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

Support \`None\` as a valid input to \`op.OpType()\` calls in the onnxscript builder. This is needed for ONNX ops with optional inputs (e.g., Gemm's optional C input).

## Changes

- Add \`None\` to the \`VALUE_LIKE\` union type
- Update \`_input_to_ir_value\` to return \`ir.Value | None\` and pass \`None\` through
- Update \`_partition_inputs_attributes\` and \`call_op\` type signatures to accept \`None\`
- Add tests for \`None\` inputs with standard and custom domain ops

Also removed pylint because it is not doing much good nowadays.

### CODE REVIEW

File: onnxscript/_internal/builder.py [hunk 1]
```diff
@@ -267,6 +268,8 @@ def _input_to_ir_value(
         """
         if isinstance(value, ir.Value):
             return value
+        if value is None:
+            return value
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply onnxscript/_internal/builder.py:272]: That's fine
